apply_license.py

```diff
@@ -76,76 +75,79 @@ def match_patterns(s, patterns):
             return True
     return False
 
+def skip_path(fullpath):
+    return match_patterns(fullpath, excluded_paths)
 
-def force_include(fullpath):
-    if match_patterns(fullpath, include_patterns):
-        return True
-    return False
-
-def skip_filename(fullpath, basename):
-    if basename in exluded_files:
-        return True
-
-    if match_patterns(fullpath, exluded_patterns):
-        return True
+def skip_filename(fullpath):
+    return match_patterns(fullpath, excluded_files)
 
-    return False
+def has_defold_license(s):
+    return "Licensed under the Defold License version" in s
 
 def has_other_license(s):
-    return "Copyright" in s or "License" in s
+    return ("Copyright" in s or "License" in s) and not has_defold_license(s)
 
-def get_comment_style_for_file(fullpath):
-    ext = os.path.splitext(fullpath)[1]
-    if ext == ".go" and not fullpath.startswith("./go/"):
+def get_license_for_file(filepath):
+    ext = os.path.splitext(filepath)[1]
+    if not ext in ext_to_license:
         return None
-    return ext_to_comment.get(ext)
+    return ext_to_license[ext]
+
+def apply_license(license, contents):
+    # Preserve shebang
+    if contents.startswith("#!"):
+        firstline = contents.partition('\n')[0]
+        contents = contents.replace(firstline, "")
+        license = firstline + "\n" + license
+    return license  + "\n\n" + contents
 
+def check_ignored(path):
```

Comment:
Check if a file is in a .gitignore in the repo. Will return 0 if it is ignored, 1 otherwise